samples/nrf9160/cloud_simple/src/main.c

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <misc/reboot.h>
+#include <logging/log_ctrl.h>
+#include <lte_lc.h>
+#include <net/cloud.h>
+#include <net/socket.h>
+
+static struct cloud_backend *cloud_backend;
+static struct k_delayed_work cloud_update_routine_work;
+
+enum error_type {
+	ERROR_CLOUD,
+	ERROR_BSD_RECOVERABLE,
+	ERROR_LTE_LC,
+	ERROR_SYSTEM_FAULT
+};
+
+/**@brief Cloud simple error handler. */
+void error_handler(enum error_type err_type, int err_code)
+{
+#if !defined(CONFIG_DEBUG) && defined(CONFIG_REBOOT)
+	LOG_PANIC();
+	sys_reboot(0);
+#else
+	switch (err_type) {
+	case ERROR_CLOUD:
+		printk("Error of type ERROR_CLOUD: %d\n", err_code);
+		break;
+	case ERROR_BSD_RECOVERABLE:
+		printk("Error of type ERROR_BSD_RECOVERABLE: %d\n", err_code);
+		break;
+	case ERROR_LTE_LC:
+		printk("Error of type ERROR_LTE_LC: %d\n", err_code);
+		break;
+	case ERROR_SYSTEM_FAULT:
+		printk("Error of type ERROR_SYSTEM_FAULT: %d\n", err_code);
+	default:
+		printk("Unknown error type: %d, code: %d\n",
+			err_type, err_code);
+		break;
+	}
+
```

Comment:
Perhaps we should move the rebooting here, the error printing may be informative to the user. Reboot could happen after a few seconds delay.

Reply:
Isn't that decided by the user when setting the CONFIG_DEBUG option? If CONFIG_DEBUG is high (and CONFIG_REBOOT) the error printing will get printed and reboot avoided.

Reply:
Yep, I'm just thinking that the information could be printed also in the case of reboot when `CONFIG_DEBUG=n`.

Reply:
In the typical case where `CONFIG_DEBUG=n`, the user won't be notified of BSD errors now, it will just reboot. It should also be fixed in the asset tracker, but not in this PR.